xwiki-batchimport-postprocessors/listidentifier/src/main/java/org/xwiki/batchimport/postprocessors/ListIdentifierPostprocessor.java

```diff
@@ -93,18 +93,19 @@ public void postProcessRow(Map<String, String> data, List<String> row, int rowIn
                 PropertyInterface pi = mappedClass.get(xwikiField);
                 if (pi instanceof PropertyClass) {
                     PropertyClass prop = (PropertyClass) pi;
-                    if (prop instanceof DBListClass) {
+                    if (prop instanceof ListClass) {
                         logger
                                 .debug("Found column mapped to " + pi.getName()
-                                        + " which is a dblist, reverse mapping...");
-                        Map<String, String> values = getValuesForProperty(mappedClass, (DBListClass) prop);
+                                        + " which is a list, reverse mapping...");
+                        Map<String, String> values = getValuesForProperty(mappedClass, (ListClass) prop);
                         // find the value corresponding to the current value in the file
                         String dataInFile = data.get(xwikiField);
                         logger.debug("Value to reverse map: " + dataInFile);
                         // split with separator
                         List<String> dataToStore = new ArrayList<>();
                         for (String v : StringUtils.split(dataInFile, config.getListSeparator())) {
-                            String valueToAdd = values.get(v);
+                            String valueToAdd = lookupKey(values, v, prop);
+
                             if (StringUtils.isNotBlank(valueToAdd)) {
                                 dataToStore.add(valueToAdd);
                             }
```

Comment:
hmmm... I don't see any fallback in lookupKey() to return the value as is for the case when no key can be identified for a given value, and no else branch here either. Is this code reverting the fix that was done for BATCHIMP-26 in the previous pull request?

Reply:
My intent was that this pull request, once merged with the one related to BATCHIMP-26 would provide the expected fallback. I managed to get that result locally by merging the BATCHIMP-26 pull request commit with this one but there might be issues with the merging order, or a conflict (as mentioned by GitHub)? Or I made a mistake? We may re-add the fallback manually if it's easier?

Reply:
Indeed, I guess this is how it should work... However, apparently there are conflicts for this PR. I'll check that and merge.